### DISCUSSION TITLE
Best way to loop through for a long list of location with geo-coordinates

### DISCUSSION THREAD

**Discussion by FayzulSaimun** [omkarcloud/google-maps-scraper#81] [General]
Hello! I have a list of locations with their respective geo-coordinates. Currently, I am running the following code:

```
for city in tqdm(batch):
        query = f"{city['city']}, {city['state_name']} karaoke"
        queries = [query]
        coordinate_string = f"{city['lat']},{city['lng']}"
        data = Gmaps.places(
            queries, fields=Gmaps.ALL_FIELDS, geo_coordinates=coordinate_string, zoom=16
        )
```
I tried this way also but it didn't work
```
queries = [{'keywords': 'erie pennsylvania karaoke', 'geo_coordinates': '42.10958,-80.104055', 'zoom': 16}]
Gmaps.places(queries, fields=Gmaps.ALL_FIELDS)
```

Can the full list be given to the scraper without the loop to minimize the run time? Currently, it takes a lot of time to get 100 location results.

**Comment by Chetan11-dev** [comment]
Yes, you need to use first code block, Also, in a few months, I plan to integrate kubernetes feature into the Botasaurus Framework. Once integrated, it will be made available as an API. and you can get 100's of keywords scraped in just 10 minutes. I am working on it. So just wait few months, and then you can scrape really really fast.

**Reply by FayzulSaimun** [reply]
Thanks for this tool. It helps a lot. Also please update the docker instruction. The scraper is not properly working on Linux server.

**Comment by Chetan11-dev** [comment]
use docker-compose build and docker-compose up it runs correctly

**Comment by FayzulSaimun** [comment]
![image](https://github.com/omkarcloud/google-maps-scraper/assets/24837432/b5625783-b257-41e7-807d-7eea09065c1b)
I got this error using docker-compose up

**Comment by Chetan11-dev** [comment]
Fixed, repull the latest docker image and run

**Comment by Chetan11-dev** [comment]
Also Fayzul,

I came across your freelancing profile and noticed your expertise in web scraping, which is impressive. I'm curious to hear your thoughts on the Botasaurus Framework. Do you consider it a strong project?

**Reply by FayzulSaimun** [reply]
Hey Chetan,

I came across your Bose framework around 4-5 months ago, but I didn't get a chance to try it out. However, I recently used your Google Maps Scraper and I must say I am very impressed with your work. The tool is top-notch. Currently, I am trying to familiarize myself with Botasaurus and I am using it in a web scraper. While it is a strong project, I have noticed some issues with it. I will make note of these issues so that you can consider fixing them to make the project even stronger.

**Reply by Chetan11-dev** [reply]
Thank You!

**Reply by Chetan11-dev** [reply]
@FayzulSaimun, we have released v4 of botasaurus, I am really interested in knowing your review of botasaurus v4 as a expert web scraper.  Could you visit https://github.com/omkarcloud/botasaurus and let me know?

**Reply by FayzulSaimun** [reply]
@Chetan11-dev  Hello, I am currently using version 4 of Botasaurus. I have developed a few scrapers based on Botasaurus for my clients, and one of them is being used by several users. The best features are the parallel operations and proxy support. I also like the UI creation features and I am planning to give it a try soon.

**Reply by Chetan11-dev** [reply]
Yes Sure, the humaness of the Scrapers have been really upped as well, and is the best in js and python universe. Also let me know about the UI Feautres review.

**Comment by Vasile-Marian-Danci** [comment]
Hello, I am really interested in doing this too. I have a list of 50.000 locations, and 300 cities ready to be scrapped. However I need some help in making the right modifications in the code. If you want, I think it would be great to hop on a call on Discord and see what we can do together.

**Reply by FayzulSaimun** [reply]
Hey, if you're interested we can connect on Discord and can discuss more.

**Reply by Vasile-Marian-Danci** [reply]
Great, I am #boxi8220. I just had a striking good idea to implement, should not be hard but I believe it would increase drastically the chances of finding ideal leads.

**Comment by Kten987** [comment]
I don't know if this is what you're looking for, but I think you're trying to crawl data for many different coordinates. If so, you can refer to the following:

1. Add a for loop at line 326 of gmaps.py
     ![image](https://github.com/omkarcloud/google-maps-scraper/assets/82595121/341ddeeb-b93d-4b21-a575-80194786f8d3)

2. Add your input variable geo_coordinates in main.py
For example
geo_coordinates = ["10.060336116445198, 105.7793725804415",
"10.041776517690092, 105.79192278333913",
"10.04857477707304, 105.78772157139154",
"10.046438333359838, 105.78541529587908"]

**Reply by FayzulSaimun** [reply]
@Kten987  Thanks. I will try this method.

**Reply by Kten987** [reply]
@FayzulSaimun forgot you will also have to change the geo coordinates variable to list[str] instead of just [str]. At line 292 of gmaps.py
![image](https://github.com/omkarcloud/google-maps-scraper/assets/82595121/f94ba427-5c29-4587-a256-0e507e2148aa)

**Comment by Chetan11-dev** [comment]
@FayzulSaimun Do you know code/api that when given a location like Dhaka returns a list of multiple geo-coordinates for it?

**Reply by FayzulSaimun** [reply]
I am using  [Worldcitydb.com](https://www.worldcitydb.com/) to get the location and their geo-coordinates.

**Reply by Kten987** [reply]
I use platform.here.com to get geo-coordinates for the location I need. However, using it for free will only get 1000 requests a day. If you use more, you have to pay or you can register for many different accounts.

**Reply by Chetan11-dev** [reply]
Thanks for letting me know.
